app/hooks/useProfileScreen.ts

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from "react"
+import { supabase } from "@/utils/supabaseClient"
+import { TxKeyPath } from "@/i18n"
+import { useStores } from "@/models"
+import { GoogleSignin } from "@react-native-google-signin/google-signin"
+
+interface UseProfileScreenReturn {
+  displayName: string
+  isEditing: boolean
+  error: TxKeyPath | undefined
```

Comment:
```suggestion
  error?: TxKeyPath
```

Funcionaria igual? creo que es mas rapido y se sobreentiende en cuanto a legibilidad